blog-posts/boosting-developer-productivity-with-github-actions.md

```diff
@@ -0,0 +1,266 @@
+
+**How does it solve the problem?**
+The workflow that we are using to do this is a mixture of publicly available actions and a shell script to extract the change in the `CHANGELOG.md` file. It get's triggered whenever the repository receives a new `tag` and it automatically creates the release in GitHub. This is just one more thing off our minds, one less thing to worry about when the release panic hits.
+
+![A release created by github-actions](./images/github-actions/release.jpg)
+
+#### "Stale issues"
+As part of the issue management we need to keep track of old issues.
+
+**What problem does it solve?**
+As our repository grows and evolves so does the number of issues that accumulate. Over time some of these issues become obsolete.
+
+**How does it solve the problem?**
+Using a scheduled GitHub Workflow that runs once every night we mark "old" issues (90 days without interaction) as stale. This workflow uses the official `actions/stale` action. If a stale issue is not being interacted with within 5 days it will get closed automatically. If an issue gets marked as stale and the team agrees that it's still relevant during the next refinement meeting, we remove the `stale` label.
+
+![github-actions marking an issue as stale](./images/github-actions/stale-issue.jpg)
+
+#### "Release comments"
+We use keywords in Pull Requests like `closes`, `fixes`, or `resolves` to automatically close issues that are done from a development point of view.
+
+**What problem does it solve?**
+With issues being closed when a corresponding Pull Request got closed it's sometimes hard to keep track of the release that a certain fix or feature got published with.
+
+**How does it solve the problem?**
+This GitHub Workflow leaves a comment on released issues that informs any subscriber about the specific release that shipped the changes necessary to close the issue. This is done using a custom bash script that reads the CHANGELOG and extracts all issues that got published as part of the latest release. This list of released issues is then being passed to a custom action that uses GitHub GraphQL API to leave comments on these issues.
+
+![A comment by github-actions that points to the corresponding release](./images/github-actions/release-comment.jpg)
+
+#### "Housekeeping"
+Housekeeping is an important aspect of the fast-paced frontend world. We used to have an issue to remind us to update our dependencies every once in a while. As with a lot of manual processes we forgot to do this from time to time. Especially when it comes to important vulnerability fixes it’s important to keep up to date. We now have a workflow to take care of that once a week using a scheduled event. With integrated tools like "Dependabot" this workflow might be obsolete soon, but so far we enjoy the fact that we are in full control of the workflow and can tweak and configure it to our hearts content.
+
+## Driven by laziness
```

Comment:
yeah that's a cool summary of the motivation for GitHub Actions or automation in general 😄 It should also include reliability because human will fail those tasks randomly for sure.

Reply:
I can't think of a nice way to write this right now...
Also, by mentioning the "human forgetfulness", I'm kind of indirectly mentioning the reliability of machines. Could that be enough?

Reply:
you refer to that:

> humans simply forget to do things from time to time.


I'm good with that and too be honest I did not see that before. I think it's enough. Thanks.